tests/integration/models/test_evaluate_constitutive_laws.py

```diff
@@ -0,0 +1,252 @@
+            7e-3 + (0.8 - 7e-3) * (1 - 0.8) / (11.11 * pp.GIGA) * 2,
+            None,
+        ),
+        (
+            setup_utils.Thermoporomechanics,
+            "matrix_porosity",
+            # phi_0 + (alpha - phi_ref) * (1 - alpha) / bulk * p
+            # - (alpha - phi_0) * thermal expansion * T
+            #  Only pressure and temperature, not div(u), is included in this test.
+            (
+                7e-3
+                + (0.8 - 7e-3) * (1 - 0.8) / (11.11 * pp.GIGA) * 2
+                - (0.8 - 7e-3) * 1e-5 * 3
+            ),
+            None,
+        ),
+        (
+            setup_utils.MomentumBalance,
+            "bulk_modulus",
+            # \lambda + 2/3 \mu
+            11.11 * pp.GIGA + 2 / 3 * 16.67 * pp.GIGA,
+            None,
+        ),
+        (
+            setup_utils.MomentumBalance,
+            "shear_modulus",
+            # \mu
+            16.67 * pp.GIGA,
+            None,
+        ),
+        (
+            setup_utils.MomentumBalance,
+            "youngs_modulus",
+            # \mu (3 \lambda + 2 \mu) / (\lambda + \mu)
+            16.67 * (3 * 11.11 + 2 * 16.67) / (11.11 + 16.67) * pp.GIGA,
+            None,
+        ),
+        (
+            _add_mixin(pp.constitutive_laws.BartonBandis, setup_utils.MomentumBalance),
+            "elastic_normal_fracture_deformation",
+            # (-normal_traction) * maximum_closure /
+            #    (normal_stiffness * maximum_closure + (-normal_traction))
+            (1 * 0.1683) / (1529 * 0.1683 + 1),
+            1,
+        ),
+        (
+            _add_mixin(c_l.CubicLawPermeability, setup_utils.MassBalance),
+            "permeability",
+            1e-20,
+            2,
+        ),
+        (
+            _add_mixin(c_l.CubicLawPermeability, setup_utils.MassBalance),
+            "permeability",
+            0.01**3 / 12,
+            1,
+        ),
+        (
+            _add_mixin(c_l.CubicLawPermeability, setup_utils.MassBalance),
+            "permeability",
+            0.01**4 / 12,
+            0,
+        ),
+    ],
+)
+def test_evaluated_values(model, method_name, expected, dimension):
+    """Test that the value of the parsed operator is as expected.
+
+    Parameters:
+        model: Model class to be tested. Likely composed of several mixins.
+        method_name: The specific method to be tested.
+        expected: The expected value from the evaluation.
+        dimension: Dimension on which the method will be invoked. If None, all
+            dimensions in the MixedDimensionalGrid will be considered.
+
+    """
+    # The thermoporoelastic model covers most constitutive laws, so we use it for the
+    # test.
+    # Assign non-trivial values to the parameters to avoid masking errors.
+    solid = pp.SolidConstants(setup_utils.granite_values)
+    fluid = pp.FluidConstants(setup_utils.water_values)
+    params = {
+        "material_constants": {"solid": solid, "fluid": fluid},
+        "fracture_indices": [],
```

Comment:
Doesn't this give a zero-fracture domain? If so, how can you test lower dimensions?

Reply:
I think you are right. Deleted the line.